mithril-aggregator/src/services/snapshotter/appender.rs

```diff
@@ -0,0 +1,67 @@
+use anyhow::{anyhow, Context};
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+use mithril_common::StdResult;
+
+use crate::services::SnapshotError;
+
+/// Define multiple ways to append content to a tar archive.
+pub(super) trait TarAppender {
+    fn append<T: Write>(&self, tar: &mut tar::Builder<T>) -> StdResult<()>;
+}
+
+pub(super) struct AppenderDirAll {
+    pub db_directory: PathBuf,
+}
+
+impl TarAppender for AppenderDirAll {
```

Comment:
We may move some tests to this file, especially those with particular cases (same files, sub folder in a folder already added, ...)